lisa/tools/lsvmbus.py

```diff
@@ -0,0 +1,148 @@
+            self.rel_id, self.target_cpu = (
+                matched_channel.group("rel_id"),
+                matched_channel.group("cpu"),
+            )
+
+
+class VmBus:
+    def __init__(self, vmbus_raw: str) -> None:
+        self.parse_vmbus(vmbus_raw)
+
+    def parse_vmbus(self, raw_str: str) -> Any:
+        matched_vmbus = PATTERN_VMBUS.match(raw_str)
+        matched_device = PATTERN_DEVICE.match(raw_str)
+        channel_vp_map_list: List[ChannelVPMap] = []
+        raw_channels = re.finditer(PATTERN_CHANNEL, raw_str)
+        for channel in raw_channels:
+            matched_channels = PATTERN_CHANNEL.match(channel.group())
+            if matched_channels:
+                channel_vp_map = ChannelVPMap(matched_channels.group())
+            channel_vp_map_list.append(channel_vp_map)
+        if matched_vmbus:
+            self.id = matched_vmbus.group("index")
+            self.vmbus_name = matched_vmbus.group("name")
+            self.class_id = matched_vmbus.group("class_id")
+        if matched_device:
+            self.device_id = matched_device.group("device_id")
+        self.channel_vp_map = channel_vp_map_list
+
+
+class Lsvmbus(Tool):
+    _lsvmbus_repo = (
+        "https://raw.githubusercontent.com/torvalds/linux/master/tools/hv/lsvmbus"
+    )
+
+    @property
+    def command(self) -> str:
+        return self._command
+
+    def can_install(self) -> bool:
+        return True
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self._command = "lsvmbus"
+        self._vmbuses: List[VmBus] = []
+
+    def _check_exists(self) -> bool:
+        _exists = super()._check_exists()
+        if not _exists:
+            previous_command = self._command
+            self._command = "$HOME/.local/bin/lsvmbus"
+            _exists = super()._check_exists()
+            if not _exists:
+                self._command = previous_command
+        return _exists
+
+    def install(self) -> bool:
+        linux_os: Linux = cast(Linux, self.node.os)
+        package_name = ""
+        if isinstance(linux_os, Redhat):
+            package_name = "hyperv-tools"
+        elif isinstance(linux_os, Suse):
+            package_name = "hyper-v"
+        elif isinstance(linux_os, Debian):
+            package_name = "linux-cloud-tools-common"
+        if package_name:
+            linux_os.install_packages(package_name)
+        else:
+            wget_tool = self.node.tools[Wget]
+            file_path = wget_tool.get(self._lsvmbus_repo, "$HOME/.local/bin")
+            # make the download file executable
+            self.node.execute(f"chmod +x {file_path}")
+            self._command = file_path
+        return self._check_exists()
+
+    def get_vmbuses(self, force_run: bool = False) -> List[VmBus]:
+        cached_result = None
+        if (not self._vmbuses) or force_run:
+            cached_result = self.run("-vv")
+            if cached_result.exit_code != 0:
+                self._command = "sudo lsvmbus"
```

Comment:
It needs a check if the path is full path like `$HOME/.local/bin/lsvmbus`. The logic can like 
```
if not self._command.startswith("sudo")
    self._command = f"sudo {self._command}"
```